packages/x-components/src/x-modules/queries-preview/utils/get-hash-from-query-preview.ts

```diff
@@ -0,0 +1,40 @@
+import { md5 } from 'js-md5';
+import { QueryPreviewInfo, QueryPreviewItem } from '../store/index';
+
+/**
+ * Creates a query hash to store a QueryPreview, so the same query
+ * with different filters can be saved more than once in the state.
+ *
+ * @param queryPreview - The {@link QueryPreviewItem | QueryPreviewItem} used in the request.
+ * @returns A unique id that will be used as a key to store the QueryPreviewItem in the state.
+ */
+export const getHashFromQueryPreviewItem = (queryPreview: QueryPreviewItem): string => {
+  let queryPreviewFilters = '';
+  if (queryPreview.request.filters) {
+    const queryPreviewFiltersKeys = Object.keys(queryPreview.request.filters);
+    queryPreviewFiltersKeys.forEach(key => {
+      queryPreview.request.filters![key].forEach(filter => {
+        queryPreviewFilters = queryPreviewFilters.concat('-' + filter.id.toString());
+      });
+    });
+  }
```

Comment:
You can refactor this to:

```suggestion
  const queryPreviewFilters = Object.values(queryPreview.request.filters)
  .flat()
  .map(filter => filter.id.toString())
  .join('-');
```